Lib/asyncio/proactor_events.py

```diff
@@ -635,7 +635,13 @@ def __init__(self, proactor):
         self._make_self_pipe()
         if threading.current_thread() is threading.main_thread():
             # wakeup fd can only be installed to a file descriptor from the main thread
-            signal.set_wakeup_fd(self._csock.fileno())
+            oldfd = signal.set_wakeup_fd(self._csock.fileno())
+            if oldfd != -1:
+                warnings.warn(
+                    'Overriding signal wakeup file '
+                    'descriptor for loop signal handlers',
```

Comment:
I no longer like string literals split across lines like this, it makes phrase searching harder. Also I'm trying to clarify the warning.
```suggestion
                    "Signal wakeup fd was already set",
```

Reply:
Done in https://github.com/python/cpython/pull/96807/commits/bc7b849d3b29b643f0352441142e95fb0a72bcbe.